packages/gateway/src/metrics.js

```diff
@@ -91,23 +102,81 @@ export async function metricsGet(request, env, ctx) {
     `# TYPE nftgateway_requests_total counter`,
     ...env.ipfsGateways.map(
       (gw) =>
-        `nftgateway_requests_total{gateway="${gw}",env="${env.ENV}"} ${
-          metricsCollected.ipfsGateways[gw].totalSuccessfulRequests +
-          metricsCollected.ipfsGateways[gw].totalFailedRequests
-        }`
+        `nftgateway_requests_total{gateway="${gw}",env="${env.ENV}"} ${totalResponsesPerGateway[gw]}`
     ),
     `# HELP nftgateway_successful_requests_total Total successful requests performed to each gateway.`,
     `# TYPE nftgateway_successful_requests_total counter`,
     ...env.ipfsGateways.map(
       (gw) =>
-        `nftgateway_successful_requests_total{gateway="${gw}",env="${env.ENV}"} ${metricsCollected.ipfsGateways[gw].totalSuccessfulRequests}`
+        `nftgateway_successful_requests_total{gateway="${gw}",env="${
+          env.ENV
+        }"} ${
+          metricsCollected.ipfsGateways[gw].totalResponsesByStatus[
+            HTTP_SUCCESS_CODE
+          ] || 0
+        }`
     ),
     `# HELP nftgateway_failed_requests_total Total failed requests performed to each gateway.`,
     `# TYPE nftgateway_failed_requests_total counter`,
     ...env.ipfsGateways.map(
       (gw) =>
-        `nftgateway_failed_requests_total{gateway="${gw}",env="${env.ENV}"} ${metricsCollected.ipfsGateways[gw].totalFailedRequests}`
+        `nftgateway_failed_requests_total{gateway="${gw}",env="${env.ENV}"} ${
+          totalResponsesPerGateway[gw] -
+          (metricsCollected.ipfsGateways[gw].totalResponsesByStatus[
+            HTTP_SUCCESS_CODE
+          ] || 0)
+        }`
     ),
+    `# HELP nftgateway_failed_requests_by_status_total Total failed requests by status code performed to each gateway.`,
+    `# TYPE nftgateway_failed_requests_by_status_total counter`,
+    ...env.ipfsGateways
+      .map((gw) => {
+        return Object.keys(
+          metricsCollected.ipfsGateways[gw].totalResponsesByStatus
+        )
+          .filter(
+            (s) =>
+              s !== HTTP_SUCCESS_CODE.toString() &&
+              metricsCollected.ipfsGateways[gw].totalResponsesByStatus[s]
+          )
+          .map(
+            (status) =>
+              `nftgateway_failed_requests_by_status_total{gateway="${gw}",env="${
+                env.ENV
+              }",status="${status}"} ${
+                metricsCollected.ipfsGateways[gw].totalResponsesByStatus[
+                  status
+                ] || 0
+              }`
+          )
+          .join('\n')
+      })
+      .filter((e) => !!e),
```

Comment:
this filter is annoying, but was the sanest way of not having empty lines when status errors only exist for a subset of known gateways

Reply:
Ya, I think not including success is unnecessarily complicated.

Reply:
This is not because of success, but because if there is for example status `429` for gateway x, but not for gateway `y`, we wend up with an empty entry

Reply:
Oh ok sorry!